src/amber/router/params.cr

```diff
@@ -1,20 +1,104 @@
+require "http"
+require "./parsers/*"
+require "./file"
+
 module Amber::Router
-  class Params < Hash(String, String)
-    alias KeyType = String | Symbol
-    property files = {} of String => Amber::Router::Files::File
+  module Types
+    alias Key = String | Symbol
+    alias Files = Hash(String, Amber::Router::File)
+    alias Params = Hash(String, String)
+  end
+
+  class Params
+    TYPE_EXT_REGEX   = Amber::Support::MimeTypes::TYPE_EXT_REGEX
+    URL_ENCODED_FORM = "application/x-www-form-urlencoded"
+    MULTIPART_FORM   = "multipart/form-data"
+    APPLICATION_JSON = "application/json"
+
+    getter empty = Types::Params.new
```

Comment:
I don't like calling this empty. 

Also I think we should use HTTP::Params instead of Hash(String, String).
After all that was the reasoning behind adding it as a property it instead of inheriting from it in the first place.